src/main/java/knightminer/inspirations/utility/block/BlockPipe.java

```diff
@@ -1,292 +1,266 @@
+			DOWN, UP, NORTH, SOUTH, WEST, EAST
+	};
+
 	public BlockPipe() {
-		super(Material.IRON);
-		this.setHardness(3.0F);
-		this.setResistance(8.0F);
-		this.setSoundType(SoundType.METAL);
-		this.setDefaultState(this.blockState.getBaseState()
-				.withProperty(FACING, EnumFacing.NORTH)
-				.withProperty(NORTH, false)
-				.withProperty(EAST, false)
-				.withProperty(SOUTH, false)
-				.withProperty(WEST, false)
-				.withProperty(UP, false)
-				.withProperty(DOWN, false)
-				.withProperty(HOPPER, false));
-		this.setCreativeTab(CreativeTabs.REDSTONE);
+		super(Block.Properties
+				.create(Material.IRON)
+				.hardnessAndResistance(3.0F, 8.0F)
+				.sound(SoundType.METAL)
+		);
+		this.setDefaultState(this.getStateContainer().getBaseState()
+				.with(FACING, Direction.NORTH)
+				.with(NORTH, false)
+				.with(EAST, false)
+				.with(SOUTH, false)
+				.with(WEST, false)
+				.with(UP, false)
+				.with(DOWN, false)
+				.with(HOPPER, false));
 	}
 
-
-	/* Block state settings */
+	/* IHidable */
 
 	@Override
-	protected BlockStateContainer createBlockState() {
-		return new BlockStateContainer(this, FACING, NORTH, EAST, SOUTH, WEST, UP, DOWN, HOPPER);
+	public boolean isEnabled() {
+		return Config.enablePipe.get();
 	}
 
 	@Override
-	public IBlockState getStateFromMeta(int meta) {
-		return this.getDefaultState().withProperty(FACING, EnumFacing.getFront(meta & 7));
+	public void fillItemGroup(@Nonnull ItemGroup group, NonNullList<ItemStack> stacks) {
+		if(shouldAddtoItemGroup(group)) {
+			super.fillItemGroup(group, stacks);
+		}
 	}
 
+	/* Block state settings */
+
 	@Override
-	public int getMetaFromState(IBlockState state) {
-		return state.getValue(FACING).getIndex();
+	protected void fillStateContainer(StateContainer.Builder<Block, BlockState> builder) {
+		builder.add(FACING, NORTH, EAST, SOUTH, WEST, UP, DOWN, HOPPER);
 	}
 
+	@Nonnull
 	@Override
-	public IBlockState withRotation(IBlockState state, Rotation rot) {
-		return state.withProperty(FACING, rot.rotate(state.getValue(FACING)));
+	public BlockState rotate(@Nonnull BlockState state, Rotation rot) {
+		return state.with(FACING, rot.rotate(state.get(FACING)));
 	}
 
+	@Nonnull
 	@Override
-	public IBlockState withMirror(IBlockState state, Mirror mirror) {
-		return state.withRotation(mirror.toRotation(state.getValue(FACING)));
+	public BlockState mirror(@Nonnull BlockState state, Mirror mirror) {
+		return state.with(FACING, mirror.mirror(state.get(FACING)));
 	}
 
+	@Nonnull
 	@Override
-	public int damageDropped(IBlockState state) {
-		return 0;
+	public BlockState updatePostPlacement(@Nonnull BlockState state, Direction neighFacing, BlockState neighState, IWorld world, BlockPos pos, BlockPos neighPos) {
+		Direction facing = state.get(FACING);
+		BlockState offsetState = world.getBlockState(pos.offset(facing));
+
+		// We only need to check the one side that updated.
+		return state
+				.with(HOPPER, offsetState.getBlock() instanceof HopperBlock &&
```

Comment:
Something feels off about this, the hopper state should only be modified if our facing is the same as the side being updated, its only relevant for the side the pipe looks into.